### ISSUE TITLE
no data

### ISSUE THREAD

**Issue by ithowest** [danxfisher/MeetEasier#1]
Hi,

cool project!!! But I seem to be failing somewhere. I'm a sysadm, not a dev guy :)
The webserver runs and I can see the room lists. I've created a user and gave this user full access to ALL room mailboxes in our office365 tenant. 
In the  nodejs logging it finds the room mailboxes but I keep getting the loading button? Any ideas?

Thx
Koen

**Comment by danxfisher** [comment]
I believe this could be related to a change I made and pushed in documentation prior to actually implementing it, heh.  In the `config/auth.js`, for the "Domain", don't include the .com, .co.uk, etc.  I accidentally pushed the change to that file before I made a change to the code.  Let me know if that helps.

**Comment by JoanneBlomert** [comment]
Hi Dan, also having an issue. Installed as per the instructions and get a spinning icon in the middle of the screen that doesn't go away. Also a sysadmin and not a developer. Tried the config/auth.js edit you mentioned about but no go. 
![meet easier](https://user-images.githubusercontent.com/33734535/32915610-b95ef2a6-cad6-11e7-9487-65db6bd4390f.png)

**Comment by danxfisher** [comment]
@JoanneBlomert could you hit F12 in the browser to bring up the developer window and copy/paste what you see in the "Console" section.  Also, anything that you can copy/paste from the server terminal window would be helpful, too.  Thanks!

**Comment by RNDnick** [comment]
Hi, I have the same issue as above, the output from the Console section is this:
socket connect at: http://localhost:8080

**Comment by JoanneBlomert** [comment]
Hi Dan, thanks for the quick reply. Was able to get this resolved. I didn’t have the room list built in Office 365 so that’s what the problem was.

Thank you,

Joanne Blomert
Helpdesk / SharePoint Administrator

The Association of Professional Engineers and Geoscientists of Alberta (APEGA)
1500 Scotia One, 10060 Jasper Ave.
Edmonton, AB T5J 4A2
PH 587-489-1538
TOLL-FREE 800-661-7020
FAX 780-425-1722

**Comment by Gavinr98** [comment]
Having the same issue, on-prem Exchange just get the spinning wheel. Get this in the dev console ---socket connect at: http://localhost:8080

**Comment by danxfisher** [comment]
@Gavinr98 your issue may be because the XMLHttpRequest includes accents (é,à,ç, ...).  Do any of your meetings or names include accented characters?

**Comment by danxfisher** [comment]
@RNDnick I would check this following:

- The service account you put in the `auth.js` file must have full access to all room mailboxes
- The service account must also be an account/mailbox on Exchange online
- The roomlists must be in Exchange online (cloud)
    
The page will also not load if any rooms/roomlists on the same domain are *not* in Exchange online.  Check those things first.  Thanks.  I will try and update the code with a check to see if anything errors, and skip it.

**Comment by Gavinr98** [comment]
No, no accents in the account. I do also see this in the ssh session after I refreshed the page. I did make sure that the account has full permissions on each calendar as well.

Error in calling service, error code:401
server: Microsoft-IIS/8.5
request-id: d08231b1-6531-4218-88b4-0f4f1cbb6a9d
www-authenticate: Negotiate, NTLM
x-powered-by: ASP.NET
x-feserver: EXCHANGESERVERNAME
date: Fri, 17 Nov 2017 22:47:08 GMT
content-length: 0

**Comment by Gavinr98** [comment]
I do also see that it is pulling in the Room List, just does not display anything and the wheel just spins.

**Comment by danxfisher** [comment]
@Gavinr98 ah this could be an IIS issue.  I have it running on IIS but it is using a module called IISNode.  I might need to test this on a server without it to see if it's possible.  I'm going to open a new issue for it.

**Comment by RNDnick** [comment]
Hi @danxfisher, thanks for replying. I'm sure all of your criteria is met, however, we are on Hybrid Exchange and some of those rooms were created on Exchange on-premise before being moved to Online. Do we think this could mnake a difference?

**Comment by danxfisher** [comment]
@RNDnick it shouldn't as I think that's how my rooms were created.  Do you have multiple domains by chance?

**Comment by RNDnick** [comment]
Multiple domain, yes and just noticed one has an email address with the .onmicrosoft.com part. Do we think this will be the issue?

**Comment by RNDnick** [comment]
I've changed the primary address on that room anyways, though still doesn't seem to have made a difference.

**Comment by ithowest** [comment]
Hi, 
In my case I had to change the primary address of all my room mailboxes to the same domain name. 
I had a couple of room mailboxes with the onmicrosoft.com suffix. That solved the configuration issue, but the api return in 9/10 requests "null" and in 1/10 refreshes I receive the data and the dashboard works. 
@RNDnick what do you get in the console of your browser?

**Comment by RNDnick** [comment]
@ithowest socket connect at: http://localhost:8080
though the bottom of the output on the web server does include a Null:
Body:
  {  __prefix: 's',
      GetRoomListsResponse:
          {  __prefix: 'm',
             __type: 'GetRoomListsResponse',
             ResponseClass: 'Success',
             ResponseCode: 'NoError',
             RoomLists: null } } }

**Comment by miltongoh** [comment]
Hi,

Try installing this on my Ubuntu 16.04 VM and have a couple of issues, not sure if you have encountered this before.

![meeteasier_issue_01](https://user-images.githubusercontent.com/6261496/33237609-6e59ad5c-d2b3-11e7-80fc-97ef3af1e36c.PNG)

The moment I hit the URL (http://<server ip>:8080/ and the top navigation loads for a bit and it just disappear leaving the black background only.

**Comment by Gtaucher** [comment]
Hello,
I'm having this problem.

Failed to load resource: net::ERR_CONNECTION_REFUSED
polling-xhr.js:263 GET http://localhost:8080/socket.io/?EIO=3&transport=polling&t=MD5IkiX 0 ()
i.create @ polling-xhr.js:263
i @ polling-xhr.js:165
o.request @ polling-xhr.js:92
o.doPoll @ polling-xhr.js:122
r.poll @ polling.js:118
r.doOpen @ polling.js:63
r.open @ transport.js:80
r.open @ socket.js:245
r @ socket.js:119
r @ socket.js:28
r.open.r.connect @ manager.js:226
(anonymous) @ manager.js:540
setTimeout (async)
r.reconnect @ manager.js:530
r.maybeReconnectOnOpen @ manager.js:208
(anonymous) @ manager.js:250
r.emit @ index.js:133
r.onError @ socket.js:685
(anonymous) @ socket.js:276
r.emit @ index.js:133
r.onError @ transport.js:67
(anonymous) @ polling-xhr.js:128
r.emit @ index.js:133
i.onError @ polling-xhr.js:309
(anonymous) @ polling-xhr.js:256
setTimeout (async)
hasXDR.n.onreadystatechange @ polling-xhr.js:255
polling-xhr.js:263 GET http://localhost:8080/socket.io/?EIO=3&transport=polling&t=MD5Ilt1 0 ()

Where can I check to find the solution?
Thank you

**Comment by sunkehappy** [comment]
@RNDnick Have you solved your problem? I have the same issue. Empty room list. But I can see the meeting room via Microsoft Outlook on Windows when trying to send meeting request. Don't know what's wrong.

**Comment by Yasvah** [comment]
I had the same issue, because I hadn't created a distribution list to sort my rooms. After adding the room it is displayed
https://technet.microsoft.com/en-us/library/ee633471(v=exchg.141).aspx

**Comment by gopihc** [comment]
> I had the same issue, because I hadn't created a distribution list to sort my rooms. After adding the room it is displayed
> https://technet.microsoft.com/en-us/library/ee633471(v=exchg.141).aspx

This worked for me..

**Comment by zkenstein** [comment]
> @ithowest socket connect at: http://localhost:8080
> though the bottom of the output on the web server does include a Null:
> Body:
> { __prefix: 's',
> GetRoomListsResponse:
> { __prefix: 'm',
> __type: 'GetRoomListsResponse',
> ResponseClass: 'Success',
> ResponseCode: 'NoError',
> RoomLists: null } } }

do you solved this?

**Comment by Yasvah** [comment]
Have you created a distribution list for the rooms?
